### PR TITLE
Add StringOutputStream

### PR DESCRIPTION
**Pull request by TheSecretSquad**
`write(int b)` is the only abstract method of `OutputStream`. The other write methods are implemented in terms of it. For this reason I didn't write tests for the `OutputStream`'s other methods. Was this the right decision?

### CODE REVIEW

File: src/main/java/toyrobotsimulator/StringOutputStream.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
+package toyrobotsimulator;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+public class StringOutputStream extends OutputStream {
```
Comment (jamesladd) [inline src/main/java/toyrobotsimulator/StringOutputStream.java]: Why not use a PrintStream?

ByteArrayOutputStream os = new ByteArrayOutputStream();
PrintStream ps = new PrintStream(os);
...
String output = os.toString("UTF8");

I'd go so far as to create a CapturingPrintStream which when constructed automatically creates the ByteArrayOutputStream. The method contents() would then do the 'return os.toString("UTF8");

File: src/test/java/toyrobotsimulator/ToyRobotSimulationTest.java [hunk 2]
```diff
@@ -1,28 +1,30 @@
...
 
 	private ToyRobotSimulation toyRobotSimulation;
-	@Mock
+	private StringOutputStream stringOutputStream;
 	private PrintStream printStream;
 	
 	private void verifySimulationReportsRunning() {
-		verify(printStream).println("Running");
+		String expectedContents = "Running" + System.lineSeparator();
+		assertEquals(expectedContents, stringOutputStream.contents());
```
Comment (jamesladd) [inline src/test/java/toyrobotsimulator/ToyRobotSimulationTest.java]: Thats the stuff :)

File: src/main/java/toyrobotsimulator/StringOutputStream.java (hunk 1 above)
Comment (TheSecretSquad) [inline src/main/java/toyrobotsimulator/StringOutputStream.java]: You mean we don't need the StringOutputStream, or do you mean to implement it using a PrintStream and ByteArrayOutputStream?

File: src/main/java/toyrobotsimulator/CapturingPrintStream.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+	public String contents() {
+		return capturedContents();
+	}
+	
+	private ByteArrayOutputStream outputStream() {
+		return ((ByteArrayOutputStream)this.out);
+	}
+	
+	private String capturedContents() {
+		try {
```
Comment (jamesladd) [inline src/main/java/toyrobotsimulator/CapturingPrintStream.java:22]: Nice.

**Pull request merged**